surfactantplugin-stringextract.py/surfactantplugin_stringextract.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import json
+import os
+from pathlib import Path
+
+import binary2strings as b2s
+from loguru import logger
+
+import surfactant.plugin
+from surfactant.sbomtypes import SBOM, Software
+
+
+@surfactant.plugin.hookimpl
+def extract_strings(filename: str, hash: str, filetype: str, min_len=4):
+    """
+    Extract ASCII strings from a binary file using binary2strings.
+    :param filename (str): The full path to the file to extract information from.
+    :param hash (str): md5 hash of the file
+    :param filetype (str): File type information based on magic bytes.
+    :param min_len (int): Minimum length of strings to be considered valid.
+    """
```

Comment:
Rather than adding a new hookspec for the new plugins, I think we can reuse the existing `extract_file_info` hook so that a new hookspec doesn't need to get added for each individual plugin. An explicit `return None` could get added, or the existing implicit `return None` will make so that the hook will run, but not create any additional metadata objects for the software entry.

```suggestion
@surfactant.plugin.hookimpl(specname="extract_file_info")
def extract_strings(sbom: SBOM, software: Software, filename: str, filetype: str):
    """
    Extract ASCII strings from a binary file using binary2strings.
    :param sbom(SBOM): The SBOM that the software entry/file is being added to. Can be used to add observations or analysis data.
    :param software(Software): The software entry associated with the file to extract information from.
    :param filename (str): The full path to the file to extract information from.
    :param filetype (str): File type information based on magic bytes.
    :param min_len (int): Minimum length of strings to be considered valid.
    """
    hash = str(software.md5)
    min_len = 4
```